prow/config/config.go

```diff
@@ -106,6 +106,15 @@ type JobConfig struct {
 	// provide their own implementation.
 	ProwYAMLGetter ProwYAMLGetter `json:"-"`
 
+	// ProwYAMLGetterNoDefault is like ProwYAMLGetter, but without default
+	// values (it does not call DefaultAndValidateProwYAML()). Its sole purpose
+	// is to allow caching of ProwYAMLs that are retrieved purely from the
+	// inrepoconfig's repo, __without__ having the contents modified by the main
+	// Config's own settings (which happens mostly inside
+	// DefaultAndValidateProwYAML()). ProwYAMLGetterNoDefault is only used by
+	// GetProwYAMLCached().
+	ProwYAMLGetterNoDefault ProwYAMLGetter `json:"-"`
```

Comment:
This function only needs to be a field of the config if you are mocking it for unit tests. Otherwise it can just be a normal method.

Reply:
done